ironfish/src/blockchain/blockchain.ts

```diff
@@ -249,11 +249,11 @@ export class Blockchain<
     return genesisHeader
   }
 
-  async open(): Promise<void> {
+  async open(options?: { upgrade?: boolean }): Promise<void> {
     if (this.opened) return
     this.opened = true
 
-    await this.db.open({ upgrade: this.forceUpgrade })
+    await this.db.open({ upgrade: options ? this.forceUpgrade : undefined })
```

Comment:
Should this take the upgrade from `options`?

Reply:
opps, fixed.